src/www/ui/api/Helper/RestHelper.php

```diff
@@ -196,4 +211,61 @@ public function copyUpload($uploadId, $newFolderId, $isCopy)
         InfoType::ERROR);
     }
   }
+
+  /**
+   * @brief A safe wrapper around plugin_find
+   *
+   * Get the FOSSology plugin from the plugin array.
+   *
+   * @param string $pluginName The required plugin
+   * @return Plugin The required plugin if found, else throws an exception.
+   * @throws \UnexpectedValueException Throws exception when plugin is not
+   *         found.
+   * @uses plugin_find()
+   */
+  public function getPlugin($pluginName)
+  {
+    $plugin = plugin_find($pluginName);
+    if (! $plugin) {
+      throw new \UnexpectedValueException(
+        "Unable to find plugin " . $pluginName);
+    }
+    return $plugin;
+  }
+
+  /**
+   * @brief Check if the token request contains valid parameters.
+   *
+   * The function checks for following properties:
+   * - The format of expiry parameter should be YYYY-MM-DD and should be +1
+   *   from now().
+   * - The length of token name should be between 0 and 40.
+   * - The scope of token should be valid.
+   *
+   * @param string $tokenExpire The expiry of token requested.
+   * @param string $tokenName   The name of the token requested.
+   * @param string $tokenScope  The scope of the token requested.
+   * @return boolean|Fossology::UI::Api::Models::Info True if all parameters
+   *         are ok, else an info.
+   */
+  public function validateTokenRequest($tokenExpire, $tokenName, $tokenScope)
+  {
+    $requestValid = true;
+    if (strtotime($tokenExpire) < strtotime("tomorrow") ||
+      ! preg_match("/^[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])$/",
+        $tokenExpire) || strtotime($tokenExpire) > strtotime("+30 days")) {
```

Comment:
configurable token expiry will be better.
may be from admin -> customize section